bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/BaseThingHandler.java

```diff
@@ -582,10 +582,10 @@ protected boolean isInitialized() {
 
     @Override
     public void bridgeStatusChanged(ThingStatusInfo bridgeStatusInfo) {
-        if (bridgeStatusInfo.getStatus() == ThingStatus.ONLINE
-                && getThing().getStatusInfo().getStatusDetail() == ThingStatusDetail.BRIDGE_OFFLINE) {
-            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE);
-        } else if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE) {
+        if (ThingStatus.ONLINE.equals(bridgeStatusInfo.getStatus())
```

Comment:
Hm, my understanding has been that the base implementation should look like:

```Java
if (bridgeStatusInfo.getStatus() == ThingStatus.OFFLINE) {
    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
}
```

Reply:
Correct. This needs to be updated (and of course the test, too). Will do tomorrow.